### PR TITLE
Set up testing

### PR DESCRIPTION
**Pull request by nyandika**
# Description

Add packages for unit testing and configure vite accordingly

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

No tests outside sample run. This PR should enable writing future tests

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by principalkelvo — COMMENTED** [review]
could we have a breakdown on how to add the tests, and how they work?

**Comment by nyandika** [comment]
> could we have a breakdown on how to add the tests, and how they work?

I have added a sample navbar component and a test for it. 
Ideally you should add test for the functionality you add could be components, could be logic, etc
To run the test we use vitest, Run `npm run test` to run the tests you add

File: src/App.tsx
Comment (Rioba-Ian) [inline src/App.tsx:1]: - are we to also register the routes in the navbar component in the app parent component as well?

Comment (nyandika) [reply src/App.tsx:1]: This was a sample to showcase tests in Navbar

Comment (Rioba-Ian) [reply src/App.tsx:1]: Then it's okay. The tests are also okay.

**Review by Rioba-Ian — APPROVED** [review]

**Pull request merged**